### ISSUE TITLE
[Bug]: We need an empty response handler

### ISSUE THREAD

**Issue by simbadmorehod** [DavideGalilei/gpytranslate#20]
### What happened?

I'm sure I'm feeding the function normal string data, the problem is that sometimes an error occurs on the server, I have little programming experience to fully analyze and fix it, but I think sometimes the library gets an empty response that can't be steamed. 

### OS / Environment

Python 3.8.5 (default, Jul 28 2020, 12:59:40)
[GCC 9.3.0] on linux

### Traceback

```python-traceback
translation = await t.translate(str(text), targetlang=str(leng))
  File "/home/simulatte/simulatte/.venv/lib/python3.8/site-packages/gpytranslate/gpytranslate.py", line 118, in translate
    (
  File "/home/simulatte/simulatte/.venv/lib/python3.8/site-packages/httpx/_models.py", line 854, in json
    return jsonlib.loads(self.text, **kwargs)
  File "/usr/lib/python3.8/json/__init__.py", line 357, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python3.8/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib/python3.8/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```

### Searched before submitting

- [X] I searched for duplicates, and didn't found anything.
- [X] I searched this bug on a search engine, and didn't found anything.

**Comment by DavideGalilei** [comment]
It's a common issue with the library, it's due to Google ratelimiting requests and giving a non-json answer which cannot get properly decoded by the library. I might add a custom exception for it, or if you have any advice, feel free to tell me :)

This is the reason why I added a proxy option, to avoid relying on a single ip address which may get temporarily blocked by google for too many requess.

**Comment by simbadmorehod** [comment]
> Это обычная проблема с библиотекой, это связано с запросами Google ratelimiting и выдачей не-json-ответа, который не может быть должным образом декодирован библиотекой. Я мог бы добавить для него собственное исключение, или, если у вас есть совет, не стесняйтесь сказать мне :)
> 
> Это причина, по которой я добавил опцию прокси, чтобы не полагаться на один IP-адрес, который может быть временно заблокирован Google для слишком большого количества запросов.

From the behavior it looks very much like a small "limit" of requests from one ip, does the proxi parameter solve the problem?

**Comment by DavideGalilei** [comment]
Well, once you use proxies, each proxy will have its own ip which will get ratelimited, so you are just extending the limit to multiple ips. I don't think I can do anything about it since it's a limit on Google's side.

**Comment by simbadmorehod** [comment]
It turns out that the number of successful requests depends on the number of proxies in the "clip" , is there a check for a positive response? Or another scenario for changing proxies? How does it work?

**Comment by DavideGalilei** [comment]
Each Translator instance has its own proxy. Refer to [examples](https://github.com/DavideGalilei/gpytranslate/tree/master/examples%2Fsync) to see how to use a proxy with the library. Though, you need to write the rotating proxy logic by yourself, since in my opinion it shouldn't be handled by the library (hence using multiple instances).

**Comment by simbadmorehod** [comment]
Yes I understand, I'm a little worried because it will be my first experience with this kind of task, while I have no idea how to detect a failed response, apparently it comes "blank"?

**Comment by DavideGalilei** [comment]
To check for errors, for now, you can catch `json.JsonDecodeError` exception. Anyway, if you really need 100% quality of work, I can suggest you to buy a subscription to paid google translate APIs, since they are reliable.

**Comment by simbadmorehod** [comment]
You see, I'm from Russia, we don't pay until it's impossible to do it for free. We are trying to find this way)

**Comment by alsoGAMER** [comment]
> You see, I'm from Russia, we don't pay until it's impossible to do it for free. We are trying to find this way)

Well, I don't know what to tell you then.

**Comment by JustLoad** [comment]
@simbadmorehod

**Comment by JustLoad** [comment]
@JustLoad ma?

**Comment by nocturn9x** [comment]
> You see, I'm from Russia, we don't pay until it's impossible to do it for free. We are trying to find this way)


As a wise man once said: _"Not my business nor my problem"_

Have a nice day!

**Comment by alsoGAMER** [comment]
Closing for inactivity.
If you need further help on this topic, don't hesitate to reopen the ticket cc @simbadmorehod.

**Comment by DavideGalilei** [comment]
Fixed in #21 (now it should rate limit way less than before). I also added a custom exception (`TranslationError`) instead of thrown JsonDecodeError.
